src/components/AssetsList/AssetsList.test.jsx

```diff
@@ -0,0 +1,608 @@
+          });
+
+          it('a date added element', () => {
+            const dateAddedElement = item.find(`#asset-date-${index}`);
+            expect(dateAddedElement).toHaveLength(1);
+            expect(dateAddedElement.text()).toEqual(defaultProps.assetsList[index].date_added);
+          });
+        });
+      });
+    });
+  });
+
+  describe('AssetsList', () => {
+    it('this.state.selectedAssetIndex is -1 by default', () => {
+      expect(wrapper.state('selectedAssetIndex')).toEqual(-1);
+    });
+
+    it('this.state.selectedAssetIndex is reset to -1 on pageChange to not selectedAssetPage', () => {
+      const newPaginationData = {
+        ...defaultProps.assetsListMetadata,
+        page: 1,
+      };
+
+      wrapper.setState({
+        selectedAssetIndex: 1,
+      });
+
+      wrapper.setProps({
+        assetsListMetadata: newPaginationData,
+      });
+
+      expect(wrapper.state('selectedAssetIndex')).toEqual(-1);
+
+      resetWrapper();
+    });
+
+    it('this.state.selectedAssetIndex is reset to to original index on pageChange to selectedAssetPage', () => {
+      const assetIndex = 1;
+
+      items.at(assetIndex).simulate('mouseDown');
+
+      wrapper.setProps({
+        selectedAsset: defaultProps.assetsList[assetIndex],
+      });
+
+      expect(wrapper.state('selectedAssetIndex')).toEqual(assetIndex);
+
+      let newPaginationData = {
+        ...defaultProps.assetsListMetadata,
+        page: 1,
+      };
+
+      wrapper.setProps({
+        assetsListMetadata: newPaginationData,
+      });
+
+      expect(wrapper.state('selectedAssetIndex')).toEqual(-1);
+
+      newPaginationData = {
+        ...defaultProps.assetsListMetadata,
+        page: 0,
+      };
+
+      wrapper.setProps({
+        assetsListMetadata: newPaginationData,
+
+      });
+
+      expect(wrapper.state('selectedAssetIndex')).toEqual(assetIndex);
+
+      resetWrapper();
+    });
+
+    it('state is cleared when an empty object is passed as a prop', () => {
+      wrapper.setProps({
+        selectedAsset: defaultProps.assetsList[1],
+      });
+
+      wrapper.setState({
+        selectedAssetIndex: 1,
+        selectedAssetPage: 0,
+      });
+
+      wrapper.setProps({
+        selectedAsset: {},
+      });
+
+      expect(wrapper.state('selectedAssetIndex')).toEqual(-1);
+      expect(wrapper.state('selectedAssetPage')).toEqual(-1);
+
+      resetWrapper();
+    });
+
+    it('this.state.selectedAssetIndex is set by clicking on a listbox option', () => {
+      const assetIndex = 1;
+
+      items.at(assetIndex).simulate('mouseDown');
+
+      expect(wrapper.state('selectedAssetIndex')).toEqual(assetIndex);
+
+      resetWrapper();
+    });
+
+    it('this.state.selectedAssetPage is set by clicking on a listbox option', () => {
+      items.at(1).simulate('mouseDown');
+
+      expect(wrapper.state('selectedAssetPage')).toEqual(0);
+
+      resetWrapper();
```

Comment:
I think these `resetWrapper()` calls could go into a `beforeEach` or `afterEach` function for this `describe` block.